### DISCUSSION TITLE
How to use the openai api service deployed by VLLM to use outlines

### DISCUSSION THREAD

**Discussion by deku0818** [outlines-dev/outlines#1035] [Q&A]
I have a qwen model that runs using the vllm openai api service,
Use models.openai in outlines
But I made a mistake
![image](https://github.com/user-attachments/assets/348a82f1-09a3-4577-93f7-214a6fefdf0a)

**Comment by rlouf** [comment]
vLLM supports structured generation in its OpenAI API (using Outlines!). Check the documentation here: https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html

**Reply by deku0818** [reply]
> vLLM supports structured generation in its OpenAI API (using Outlines!). Check the documentation here: https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html
![image](https://github.com/user-attachments/assets/ee8d0b36-6188-42e6-893f-cc795c3af83f)

I specified parameters, but the same error occurred,do I need to change the model name?

**Comment by deku0818** [comment]
Is there anyone who can help me answer this question?

**Reply by lapp0** [reply]
Could you please provide your full run script? If you're getting that error, it appears that you're using `outlines.models.openai` for a non-openai model.
